src/android/CodePushReportingManager.java

```diff
@@ -72,6 +46,34 @@ public void run() {
         });
     }
 
+    public boolean hasFailedReport() {
+        if (this.hasFailedReport == null) {
+            this.hasFailedReport = codePushPreferences.getFailedReport() != null;
+        }
+
+        return this.hasFailedReport;
```

Comment:
cool one-liner that's probably uglier than what u have but w/e:
`return this.hasFailedReport = this.hasFailedReport == null ? codePushPreferences.getFailedReport() != null : this.hasFailedReport;`